src/PointwiseFunctions/AnalyticSolutions/Poisson/MathFunction.hpp

```diff
@@ -0,0 +1,107 @@
+      ::Tags::Flux<Tags::Field, tmpl::size_t<Dim>, Frame::Inertial>,
+      ::Tags::FixedSource<Tags::Field>>;
+
+  const tnsr::I<DataType, Dim>& x;
+  const ::MathFunction<Dim, Frame::Inertial>& math_function;
+
+  void operator()(gsl::not_null<Scalar<DataType>*> field,
+                  gsl::not_null<Cache*> cache, Tags::Field /*meta*/) const;
+  void operator()(gsl::not_null<tnsr::i<DataType, Dim>*> field_gradient,
+                  gsl::not_null<Cache*> cache,
+                  ::Tags::deriv<Tags::Field, tmpl::size_t<Dim>,
+                                Frame::Inertial> /*meta*/) const;
+  void operator()(gsl::not_null<tnsr::I<DataType, Dim>*> flux_for_field,
+                  gsl::not_null<Cache*> cache,
+                  ::Tags::Flux<Tags::Field, tmpl::size_t<Dim>,
+                               Frame::Inertial> /*meta*/) const;
+  void operator()(gsl::not_null<Scalar<DataType>*> fixed_source_for_field,
+                  gsl::not_null<Cache*> cache,
+                  ::Tags::FixedSource<Tags::Field> /*meta*/) const;
+};
+}  // namespace detail
+
+template <size_t Dim>
+class MathFunction : public elliptic::analytic_data::AnalyticSolution {
+ public:
+  struct Function {
+    using type = std::unique_ptr<::MathFunction<Dim, Frame::Inertial>>;
+    static constexpr Options::String help = "The solution function";
+  };
+
+  using options = tmpl::list<Function>;
+  static constexpr Options::String help{
+      "Any solution to the Poisson equation given by a MathFunction "
+      "implementation, such as a Gaussian."};
+
+  MathFunction() = default;
+  MathFunction(const MathFunction&) = delete;
+  MathFunction& operator=(const MathFunction&) = delete;
+  MathFunction(MathFunction&&) = default;
+  MathFunction& operator=(MathFunction&&) = default;
+  ~MathFunction() override = default;
+  std::unique_ptr<elliptic::analytic_data::AnalyticSolution> get_clone()
+      const override;
+
+  MathFunction(
+      std::unique_ptr<::MathFunction<Dim, Frame::Inertial>> math_function);
+
+  const ::MathFunction<Dim, Frame::Inertial>& math_function() const {
+    return *math_function_;
+  }
+
+  /// \cond
+  explicit MathFunction(CkMigrateMessage* m);
+  using PUP::able::register_constructor;
+  WRAPPED_PUPable_decl_template(MathFunction);  // NOLINT
+  /// \endcond
+
+  template <typename DataType, typename... RequestedTags>
+  tuples::TaggedTuple<RequestedTags...> variables(
+      const tnsr::I<DataType, Dim>& x,
+      tmpl::list<RequestedTags...> /*meta*/) const {
+    using VarsComputer = detail::MathFunctionVariables<DataType, Dim>;
+    typename VarsComputer::Cache cache{get_size(*x.begin())};
+    const VarsComputer computer{x, *math_function_};
+    return {cache.get_var(computer, RequestedTags{})...};
+  }
+
+  // NOLINTNEXTLINE(google-runtime-references)
+  void pup(PUP::er& p) override;
+
+  bool operator==(const MathFunction& rhs) const;
+  bool operator!=(const MathFunction& rhs) const;
```

Comment:
Make `==` and `!=` free functions.

Reply:
Then I need to instantiate them with GenerateInstantiations, which I was hoping to avoid..

Reply:
This is actually in our style guidelines via the [C++ Core Guidelines, Section C.86](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#c86-make--symmetric-with-respect-to-operand-types-and-noexcept) (See the last bullet in that section for the explicit statement.)

Reply:
ok, fixed